_posts/2017-08-10-glpi-atoum.markdown

```diff
@@ -0,0 +1,82 @@
+---
+layout: post
+title: "GLPI migrate to atoum"
+date: 2017-08-10 10:00:00
+author: "@Grummfy"
+categories: news
+---
+
+The [GLPI project][GLPI] [recently announced][announced-en][^1] that they have migrated their unit test suite to atoum. So let's get back on
+this news. We interviewed [Johan Cwiklinski](https://github.com/trasher) and some other members of the project's official team about this migration.
+
+## 1. Can you summary the GLPI project and the usage of PHP inside it?
+
+[GLPI][GLPI] is an open source IT asset management and Service Desk software. The project has started 15 years ago and is written in PHP.
+The project was led by an association until 2015, and since then the development is headed by [Teclib](www.teclib.com).
+
+Since 2015; we are trying to rewrite and modernize the whole project, starting with the backend.
+
+## 2. When did you start using unit test for the project?
+
+Firstly, one regular contributor has tried to add unit tests into the project in 2010. But project managers did not follow him, and this was quickly abandoned.
+Then, in 2015, The same contributor tried again to add tests on the project; with much more success :)
+
+## 3. What kind of difficulties do you encounter with PHPUnit? Why did you choose atoum instead of other tools to resolve it?
+
+We didn’t really face issues with PHPUnit. This is more a choice that was dictated by atoum capabilities; like:
+
+* testing variable types (if you test an integer and get a string, this is not correct),
+* wonderful mocking system (that allows mocking PHP native functions, constants, ...),
+* use closure to test outputs, exceptions, ... (way more interesting than phpunit’s annotations),
+* concurrent run of tests (even if it has been disabled for GLPI),
+* fully isolated tests,
+* chained calls,
+* more natural way to write tests (this is maybe just my point of view - but this is really simpler to me).
+
+Last but not least; atoum is a French project at the origin; just like GLPI!
+
+## 4. Did you meet some issue during the migration from PHPUnit to atoum?
+
+Well... We had to entirely rewrite our tests. It’s more a painful than a difficult work at first; but finally, we haven't spent too much time on this rewrite work.
```

Comment:
Do we have some numbers here? Time *t* spent to rewrite *x* test suites of *y* test cases? Did they find new bugs? What about the execution time? Is it easier now to write new tests? Is it easier for newcomers to write or to maintain new test suites?